### PR TITLE
Fixing issue with named series handling in fillna

### PR DESCRIPTION
**Pull request by kozlov-alexey**

### CODE REVIEW

File: hpat/hiframes/api.py [hunk 1]
```diff
@@ -361,7 +361,7 @@ def generic(self, args, kws):
 class FillNaStrType(AbstractTemplate):
     def generic(self, args, kws):
         assert not kws
-        assert len(args) == 2
+        assert len(args) == 3
```
Comment (anmyachev) [inline hpat/hiframes/api.py]: It is necessary to correct the comment under this line.

Comment (fschlimb) [reply hpat/hiframes/api.py]: Please do so.

File: hpat/tests/test_series.py [hunk 2]
```diff
@@ -627,12 +620,16 @@ def test_impl(A):
         pd.testing.assert_series_equal(hpat_func(df.A), test_impl(df.A),
             check_names=False)
 
-    @unittest.skip('numba.errors.LoweringError - fix needed\n'
-                   'Failed in hpat mode pipeline '
-                   '(step: nopython mode backend)\n'
-                   'expecting {{i64, i64, i32*, i8*, i8*, i8*}, i8*, '
-                   '{i8*, i64, i32, i32, i64, i8*, i8*}} \n'
-                   'but got {{i64, i64, i32*, i8*, i8*, i8*}, i8*, i8*}\n')
+    def test_series_fillna_inplace1(self):
```
Comment (fschlimb) [inline hpat/tests/test_series.py]: Please add a comment what this function is testing.

Comment (kozlov-alexey) [reply hpat/tests/test_series.py]: I added some comment (there are no comments for other tests though).
This new test just adds missed check for attempting to jit inplace fillna on series with numeric data. There was a test for string series, but not numeric one.

File: hpat/hiframes/api.py (hunk 1 above)
Comment (kozlov-alexey) [reply hpat/hiframes/api.py]: It's done.

File: hpat/tests/test_series.py (hunk 2 above)
Comment (fschlimb) [reply hpat/tests/test_series.py]: That's fine. We want to improve code style and docu. So anything new should be well documented.

**Pull request merged**